tools/mtmd/clip.cpp

```diff
@@ -4344,6 +4636,44 @@ bool clip_image_preprocess(struct clip_ctx * ctx, const clip_image_u8 * img, str
                 res_imgs->grid_y = inst.grid_size.height;
             } break;
 
+        case PROJECTOR_TYPE_JINACLIP2:
+            {
+                clip_image_u8 processed_image;
+                int           sz = params.image_size;
+
+                // 1) Preserve aspect ratio: resize so that the shorter side == sz (bicubic)
```

Comment:
this whole code block is redundant. it is essentially the same logic with `PROJECTOR_TYPE_GEMMA3`